app/js/sendToChart.ts

```diff
@@ -45,20 +45,22 @@ async function generateDataset(
         label: datasetLabel,
         backgroundColor: []
     }
-
-    testLabels.forEach(function (label) {
-
+    testLabels.forEach(function (label: string) {
         let amountInRange = 0;
-
-        users.forEach(function (user) {
-            if (!user.testNotTaken && user.percentage >= minPercentage && user.percentage < maxPercentage && user.testAllocated == label) {
-                amountInRange ++;
+        users.forEach(function (user: BaseUser) {
+            for(let i = 0; i < user.results.length; i++) {
```

Comment:
this looks like it could be done as a foreach over user.results instead (if yes, it should be)

if it can't be done using a foreach let me know

Reply:
Changes have been made to make the code more readable or closer to optimum? Either solution works.